src/main/java/hudson/plugins/git/util/GitUtils.java

```diff
@@ -225,12 +242,12 @@ public static EnvVars getPollEnvironment(AbstractProject p, FilePath ws, Launche
                     }
                 }
             } else {
-                env = new EnvVars(System.getenv());
+                env = p.getEnvironment(workspaceToNode(ws), TaskListener.NULL);
```

Comment:
🐛  - why not use the TaskListener we have provided?  Also might be worth surrounding this with a try/catch since the workspace may be remote and thus subject to failure.